/*
 * Copyright 2006 Sun Microsystems, Inc.  All rights reserved.
 *
 * Permission is hereby granted, free of charge, to any person obtaining a
 * copy of this software and associated documentation files (the
 * "Software"), to deal in the Software without restriction, including
 * without limitation the rights to use, copy, modify, merge, publish,
 * distribute, and/or sell copies of the Software, and to permit persons
 * to whom the Software is furnished to do so, provided that the above
 * copyright notice(s) and this permission notice appear in all copies of
 * the Software and that both the above copyright notice(s) and this
 * permission notice appear in supporting documentation.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
 * OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
 * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT
 * OF THIRD PARTY RIGHTS. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR
 * HOLDERS INCLUDED IN THIS NOTICE BE LIABLE FOR ANY CLAIM, OR ANY SPECIAL
 * INDIRECT OR CONSEQUENTIAL DAMAGES, OR ANY DAMAGES WHATSOEVER RESULTING
 * FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN ACTION OF CONTRACT,
 * NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF OR IN CONNECTION
 * WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
 *
 * Except as contained in this notice, the name of a copyright holder
 * shall not be used in advertising or otherwise to promote the sale, use
 * or other dealings in this Software without prior written authorization
 * of the copyright holder.
 *
 */

Fixes for bugs that cause problems localizing xscreensaver:
4806728 Dialognostics should be Diagnostics
6203951 xscreensaver-lock was not properly l10ned

diff -urp -x '*~' driver/xscreensaver-demo.glade2 driver/xscreensaver-demo.glade2
--- driver/xscreensaver-demo.glade2	2006-05-10 08:48:52.368024000 -0700
+++ driver/xscreensaver-demo.glade2	2006-05-10 08:48:56.382041000 -0700
@@ -1094,7 +1094,7 @@ Available</property>
 			    <widget class="GtkCheckButton" id="verbose_button">
 			      <property name="visible">True</property>
 			      <property name="can_focus">True</property>
-			      <property name="label" translatable="yes">_Verbose Dialognostics</property>
+			      <property name="label" translatable="yes">_Verbose Diagnostics</property>
 			      <property name="use_underline">True</property>
 			      <property name="relief">GTK_RELIEF_NORMAL</property>
 			      <property name="active">False</property>
diff -urp -x '*~' driver/xscreensaver.c driver/xscreensaver.c
--- driver/xscreensaver.c	2006-05-10 08:48:52.422327000 -0700
+++ driver/xscreensaver.c	2006-05-10 08:48:56.397896000 -0700
@@ -127,6 +127,9 @@
 #include <stdio.h>
 #include <ctype.h>
 #include <X11/Xlib.h>
+#include <locale.h>
+#include <libintl.h>
+#include <X11/Xlibint.h>
 #include <X11/Xatom.h>
 #include <X11/Intrinsic.h>
 #include <X11/StringDefs.h>
@@ -1305,6 +1308,15 @@ main (int argc, char **argv)
   saver_preferences *p = &si->prefs;
   int i;
 
+#ifdef ENABLE_NLS
+  if (!setlocale (LC_ALL, ""))
+    fprintf (stderr, "locale not supported by C library\n");
+
+  bindtextdomain (GETTEXT_PACKAGE, LOCALEDIR);
+  textdomain (GETTEXT_PACKAGE);
+  bind_textdomain_codeset (GETTEXT_PACKAGE, "UTF-8");
+#endif /* ENABLE_NLS */
+
   memset(si, 0, sizeof(*si));
 
 #ifdef HAVE_SCF_SMARTCARD
